lib/puppet/type/http.rb

```diff
@@ -1,17 +1,53 @@
-Puppet::Type.newtype :http, :is_capability => true do
+Puppet::Type.newtype :http do
+  @doc = "A type for testing http connections"
+
+  ensurable do
+    defaultvalues
+    defaultto :present
+  end
+
   newparam(:name, :is_namevar => true) do
-    desc 'The name of the resource'
+    desc 'The unique name of the resource'
+  end
+
+  newparam(:uri) do
+    desc 'The URL to test'
+    defaultto "http://127.0.0.1/"
```

Comment:
this will default to exporting local hosts to other nodes which is probably wrong.